services/app/src/actions/Search.tsx

```diff
@@ -63,48 +64,62 @@ export const searchAndPlay = remoteify(function searchAndPlay(id: string, dispat
 });
 
 // TODO switch to fetch since this never loads local files
-function getSearchResults(params: SearchSettings, callback: (response: QuestSearchResponse) => void) {
+function getSearchResults(params: SearchSettings, callback: (response: any) => void) {
   return (dispatch: Redux.Dispatch<any>) => {
     // Clear previous results
     dispatch({type: 'SEARCH_REQUEST'});
 
-    const xhr = new XMLHttpRequest();
-    // TODO: Pagination / infinite scrolling
-    xhr.open('POST', AUTH_SETTINGS.URL_BASE + '/quests', true);
-    xhr.setRequestHeader('Content-Type', 'text/plain');
-    xhr.onload = () => {
-      let response: QuestSearchResponse;
-      try {
-        response = JSON.parse(xhr.responseText);
-      } catch (e) {
-        response = {error: e.toString(), hasMore: false, quests: []};
-      }
+    // TODO: Create a type for questSearchPromises
+    const questSearchPromises: any = [];
 
-      // Log silently to console, so we can see if users file feedback
-      // with recurrent search problems.
-      if (response.error) {
-        console.error(response.error);
-      }
+    const partitionQuestsPromise = fetch(AUTH_SETTINGS.URL_BASE + '/quests', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'text/plain',
+      },
+      body: JSON.stringify(params),
+    });
+
+    questSearchPromises.push(partitionQuestsPromise);
+
+    if (params.partition === 'expedition-private') {
```

Comment:
Rather than having this occur as two separate requests, I was thinking we could pass a new search parameter that says "return results across both partitions" and keep it as a single request. Something like `showprivate`

Reply:
On it